python-checks/src/test/resources/checks/uselessStatement.py

```diff
@@ -136,3 +136,19 @@ def no_issue_within_contextlib_suppress(a):
   with suppress(TypeError):
     a + ''  # OK
     return a
+
```

Comment:
Those tests look good.
What about the case you mentioned with decorators? I think it was a good catch and would be relevant to have as a dedicated ticket and possibly as a documenting test here (just showing the FP might occur in this case, without actually addressing it).